ts/redux/async_dispatcher.ts

```diff
@@ -5,30 +5,30 @@ import { Web3Wrapper } from '@0x/web3-wrapper';
 import * as _ from 'lodash';
 
 import { Dispatcher } from 'ts/redux/dispatcher';
-import { AccountState, Network, ProviderState } from 'ts/types';
+import { Network } from 'ts/types';
 import { errorReporter } from 'ts/utils/error_reporter';
 
 // NOTE: Copied from Instant
 export const asyncDispatcher = {
     fetchAccountInfoAndDispatchToStoreAsync: async (
-        providerState: ProviderState,
         dispatcher: Dispatcher,
         networkId: Network,
         shouldAttemptUnlock: boolean = false,
         shouldSetToLoading: boolean = false,
+        connector: any,
     ) => {
-        const web3Wrapper = providerState.web3Wrapper;
-        const provider = providerState.provider;
-        if (shouldSetToLoading && providerState.account.state !== AccountState.Loading) {
+        const provider = await connector.getProvider();
+        const web3Wrapper = new Web3Wrapper(provider);
```

Comment:
Would this not need to be wrapped in the Metamask sub-provider if it's an injected connector?

Reply:
Not sure i follow here

Reply:
I think Web3Wrapper requires MetaMask to be wrapped in a `MetamaskSubprovider` from `@0x/subproviders`

```
const web3Provider = new Web3Wrapper(connector === injected ? new MetamaskSubprovider(provider) : provider)
```